circuits/aes-gcm/ghash.circom

```diff
@@ -127,3 +126,34 @@ template GHASH(NUM_BLOCKS) {
     // tag[0] <== intermediate[NUM_BLOCKS-1][0];
     // tag[1] <== intermediate[NUM_BLOCKS-1][1];
 }
+
+
+// Transform the GHASH hash key to a POLYVAL hash key
+// reverse the bits of `in` and multiply `h` by x
+// 
+// h.reverse();
+// let mut h_polyval = polyval::mulx(&h);
+// let result = GHash(Polyval::new_with_init_block(&h_polyval, init_block)); 
+template TranslateHashkey() {
+    signal input in[128]; 
+    signal output out[128]; 
+
+//     signal mid[128];
+
+//     // reverse bytes
+//     for (i = 0; i < 16; i++) {
+//         for (j = 0; j < 8; j++){
+//             var IDX_FROM = 120-i*8+j;
+//             var IDX_TO = i*8+j;
+//             mid[IDX_TO] <== in[IDX_FROM];
+//         }
+//     }
+
+//     component MULX;
+//     MULX = polyval_GFMULX();
```

Comment:
This approach on the surface seems dramatically more efficient than the one currently used in Waylon's PR, because it only does this GFMULX once at the end. 

Would be good to put our heads together and find a potentially combined approach of these code paths that is less constraint consumptive.